diff -rupN celix2/dfi/private/src/json_rpc.c celix/dfi/private/src/json_rpc.c
--- celix2/dfi/private/src/json_rpc.c	2015-12-08 16:05:36.796271781 +0100
+++ celix/dfi/private/src/json_rpc.c	2015-12-08 15:22:03.697788535 +0100
@@ -172,7 +172,7 @@ int jsonRpc_call(dyn_interface_type *int
                     }
                     if (dynType_descriptorType(typedType) == 't') {
                         status = jsonSerializer_serializeJson(typedType, (void*) &ptr, &jsonResult);
-                        free(ptr);
+                        //free(ptr);
                     } else {
                         dyn_type *typedTypedType = NULL;
                         if (status == OK) {
Binary files celix2/.git/index and celix/.git/index differ
