diff --git a/libs/dfi/src/json_rpc.c b/libs/dfi/src/json_rpc.c
index 6ff8367..25e1afc 100644
--- a/libs/dfi/src/json_rpc.c
+++ b/libs/dfi/src/json_rpc.c
@@ -171,7 +171,7 @@ int jsonRpc_call(dyn_interface_type *intf, void *service, const char *request, c
 					}
 					if (dynType_descriptorType(typedType) == 't') {
 						status = jsonSerializer_serializeJson(typedType, (void*) &ptr, &jsonResult);
-						free(ptr);
+						//free(ptr);
 					} else {
 						dyn_type *typedTypedType = NULL;
 						if (status == OK) {
